### PR TITLE
String dtype

### PR DESCRIPTION
**Pull request by reuster986**
Ok, this is a big one. This PR introduces strings. The functionality is not complete, but it's enough for basic usage (to see supported operations, run `make doc` and navigate to Usage -> Strings in Arkouda -> Operations). The performance is 1-2 orders of magnitude slower than for equivalent operations on numeric types, but that's a long ways from where it started. Here are the checks I have done:

- [X] Created and ran a Chapel unit test (`test/UnitTestStringType.chpl`)
- [X] Created and ran a Python test that uses the full arkouda server (`tests/string_test.py`) and compares results against NumPy for correctness
- [X] Ran Python tests `check.py`, `operator_tests.py`, `setops_test.py`, and `groupby_test.py` to verify that non-string functionality is still correct
- [X] Verified (on our Cray XC40) that this PR did not affect the performance of non-string operations by running everything in `benchmarks` on master vs. this branch
- [X] Documented new functionality in Python and Sphinx docs

Planned additions/improvements (for later, not as a prerequisite for this PR):
* A faster sort (@mhmerrill and I are discussing burst sort with tries)
* Full documentation of Chapel code
* Regex search and group capture
* Ability to write strings array to HDF5
* Concatenation (horizontal and vertical)

### CODE REVIEW

**Comment by reuster986** [comment]
Oh, and we should probably bump the `VERSION` and create a new tag if and when this PR gets merged.

File: arkouda/strings.py [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+            msg = "segmentedIndex {} {} {} {} {}".format('pdarrayIndex',
+                                                         self.objtype,
+                                                         self.offsets.name,
+                                                         self.bytes.name,
+                                                         key.name)
+            repMsg = generic_msg(msg)
+            offsets, values = repMsg.split('+')
+            return Strings(offsets, values)
+        else:
+            return NotImplemented
```
Comment (mhmerrill) [inline arkouda/strings.py]: Should this be a raise key type error?

File: src/GenSymIO_broken.chpl [hunk 2]
```diff
@@ -0,0 +1,642 @@
+module GenSymIO {
```
Comment (mhmerrill) [inline src/GenSymIO_broken.chpl]: @reuster986 should this file be deleted?

**Comment by mhmerrill** [comment]
@reuster986 i fixed a execute permission on the `string_test.py` file

**Comment by mhmerrill** [comment]
@reuster986 please pull master into your branch and resolve conflicts.

File: arkouda/strings.py (hunk 1 above)
Comment (reuster986) [reply arkouda/strings.py]: Yes, I'll change it to raise a TypeError

File: src/GenSymIO_broken.chpl (hunk 2 above)
Comment (reuster986) [reply src/GenSymIO_broken.chpl]: Yes, will delete.

**Comment by reuster986** [comment]
@mhmerrill Completed.

**Pull request merged**